app/scripts/metamask-controller.js

```diff
@@ -136,7 +137,12 @@ module.exports = class MetamaskController extends EventEmitter {
         this.accountTracker.stop()
       }
     })
-
+     
+    // ensure accountTracker updates balances after network change
+    this.networkController.on('networkDidChange', () => {
```

Comment:
Can we merge this with the handler below on line 197?

Reply:
https://github.com/MetaMask/metamask-extension/blob/59ab595b5ea6a4e24f2048d2ed43c1181bb5aa3e/app/scripts/metamask-controller.js#L197-L199

Reply:
I put this here to keep the `accountTracker` related code together
likewise the `balanceTracker` is with its own related code

we could put them all in one handler but then i would likely move that to a different location

another reason to have them separate (if we're using `SafeEventEmitter` and not `EventEmitter`) is that the action is protected in case of an error
```js
networkController.on('networkChange', a) 
networkController.on('networkChange', b) 
networkController.on('networkChange', c) 
```
in this case a,b, c will all be run even if one throws an error
this is not the case if they are all in the handler together 

that said, i dont think NetworkController is a `SafeEventEmitter`

Reply:
Sounds reasonable 👍